src/Support/Reflector.php

```diff
@@ -92,9 +96,22 @@ public static function getParametersOfType(string $type, ReflectionFunctionAbstr
 
         return collect($parameters)
             ->map(fn (ReflectionParameter $parameter) => static::getParameterClassNames($parameter))
-            ->map(fn (array $names) => array_filter($names, fn ($name) => is_a($name, $type, true)))
-            ->reject(fn (array $names) => empty($names))
-            ->map(fn (array $names) => Arr::first($names));
+            ->flatten()
+            ->filter(fn ($class_name) => is_a($class_name, $type, true));
+    }
+
+    /** @return class-string[] */
+    public static function getClassInstanceOf(string|object $class): array
+    {
+        $reflection = new ReflectionClass($class);
+
+        $class_and_interface_names = array_unique($reflection->getInterfaceNames());
+
+        do {
+            $class_and_interface_names[] = $reflection->getName();
+        } while ($reflection = $reflection->getParentClass());
+
+        return $class_and_interface_names;
     }
```

Comment:
Remove duplicate declaration

```suggestion
    /** @return class-string[] */
    public static function getClassInstanceOf(string|object $class): array
    {
        $reflection = new ReflectionClass($class);

        $class_and_interface_names = array_unique($reflection->getInterfaceNames());

        do {
            $class_and_interface_names[] = $reflection->getName();
        } while ($reflection = $reflection->getParentClass());

        return $class_and_interface_names;
    }
```

Reply:
@DanielCoulbourne @DanielCoulbourne @skylerkatz I don't have write perms on this repo otherwise I'd push the commit myself!